### PR TITLE
[SMTToLLVM] Add support for most expressions

### PR DESCRIPTION
**Pull request by maerhart**
Only the quantifier operations are still missing.

### CODE REVIEW

File: lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp [hunk 1]
```diff
@@ -779,6 +802,194 @@ struct CheckOpLowering : public SMTLoweringPattern<CheckOp> {
...
+/// function calls depending on whether the storage APInt has a bit-width that
+/// fits in a `uint64_t`.
+/// ```
+/// Z3_sort Z3_API Z3_mk_int_sort(Z3_context c);
+///
+/// Z3_ast Z3_API Z3_mk_int64(Z3_context c, int64_t v, Z3_sort ty);
+///
+/// Z3_ast Z3_API Z3_mk_numeral(Z3_context c, Z3_string numeral, Z3_sort ty);
+/// Z3_ast Z3_API Z3_mk_unary_minus(Z3_context c, Z3_ast arg);
+/// ```
```
Comment (fabianschuiki) [inline lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp:900]: Nice!

File: lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp [hunk 2]
```diff
@@ -822,6 +1033,12 @@ void circt::populateSMTToZ3LLVMConversionPatterns(
                                        converter, patterns.getContext(),       \
                                        globals, options, APINAME,              \
                                        MIN_NUM_ARGS);
+
+#define ADD_ONE_TO_ONE_PATTERN(OP, APINAME, NUM_ARGS)                          \
+  patterns.add<OneToOneSMTPattern<OP>>(/*NOLINT(bugprone-macro-parentheses)*/  \
+                                       converter, patterns.getContext(),       \
+                                       globals, options, APINAME, NUM_ARGS);
+
```
Comment (fabianschuiki) [inline lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp:1041]: Same comment as in #6902 about potentially making these templated lambdas to avoid a macro.

**Review by fabianschuiki — APPROVED** [review]
Very nice stuff. LGTM :sunglasses:

File: lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp (hunk 2 above)
Comment (maerhart) [reply lib/Conversion/SMTToZ3LLVM/LowerSMTToZ3LLVM.cpp:1041]: AFAIK templated lambdas are a C++20 feature. If there is something similar we can use in C++17, let me know!

**Pull request merged**